functions/initialize-d365testautomationcertificates.ps1

```diff
@@ -0,0 +1,56 @@
+  <#
+  .SYNOPSIS
+  Create and configure test automation certificates
+  
+  .DESCRIPTION
+  Creates a new self signed certificate for automated testing and reconfigures the AOS Windows Identity Foundation configuration to trust the certificate
+  
+  .PARAMETER CertificateFileName
+  Filename to be used when exporting the cer file
+  
+  .PARAMETER PrivateKeyFileName
+  Filename to be used when exporting the pfx file
+
+  .EXAMPLE
+  Initialize-D365TestAutomationCertificate 
+
+  This will generate a certificate for issuer 127.0.0.1 and install it in the trusted root certificates and modify the wif.config of the AOS to include the thumbprint and trust the certificate
+  #>
+  function Initialize-D365TestAutomationCertificate {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $false, Position = 1)]
+        [string]$CertificateFileName = "TestAuthCert.cer",
+
+        [Parameter(Mandatory = $false, Position = 2)]
+        [string]$PrivateKeyFileName = "TestAuthCert.pfx",
+
+        [Parameter(Mandatory = $false, Position = 3)]
+        [Security.SecureString]$Password = (ConvertTo-SecureString -String "Password1" -Force -AsPlainText),
+
+        [Parameter(Mandatory = $false, Position = 4)]
+        [string]$WindowsKitFolder = "C:\Program Files (x86)\Windows Kits\10\bin\x64"
+    )
+
+    if (!$Script:IsAdminRuntime) 
+    {
+        Write-Host "The cmdlet needs administrator permission (Run As Administrator) to be able to update the configuration. Please start an elevated session and run the cmdlet again." -ForegroundColor Yellow
+        Write-Error "Elevated permissions needed. Please start an elevated session and run the cmdlet again." -ErrorAction Stop
+    }
+
+    try 
+    {
+        # Create the certificate and place it in the right stores
+        $X509Certificate = New-D365SelfSignedCertificate -CertificateFileName $CertificateFileName -PrivateKeyFileName $PrivateKeyFileName -Password $Password -WindowsKitFolder $WindowsKitFolder
+        
```

Comment:
Use Test-PSFFunctionInterrupt to test if the function was stopped with the stop-psffunction -stepsupward 1